### ISSUE TITLE
Non-blocking WaitForUpdatesEx

### ISSUE THREAD

**Issue by thib-ack** [vmware/govmomi#739]
Hi,

I'm trying to fetch the VSphere updates from server, **in a non-blocking way**.

I tried `methods.CheckForUpdates(ctx, client.Client, request)` and it works like a charm, however the doc says:

> Deprecated. As of vSphere API 4.1, use WaitForUpdatesEx with a maxWaitSeconds of 0.

But when I try with this setup, it works in a blocking way (my param 0 seems to be ignored):

```go
var waitOptions = types.WaitOptions{
	// Here I set 0. Doc says : 
	// A value of 0 causes WaitForUpdatesEx to do one update calculation 
	// and return any results. This behavior is similar to CheckForUpdates.
	MaxWaitSeconds: 0,
}
req := types.WaitForUpdatesEx{
	This:    collector.Reference(),
	// My version. "" the first time, then I use the version returned by server
	Version: version,
	Options: &waitOptions,
}
res, err := methods.WaitForUpdatesEx(ctx, client.Client, &req)
// Block until something has changed
```

I'm wondering if the **omitempty** tag on the MaxWaitSeconds field is not the cause of the problem ? 

```go
type WaitOptions struct {
	DynamicData
	MaxWaitSeconds   int32 `xml:"maxWaitSeconds,omitempty"`
	MaxObjectUpdates int32 `xml:"maxObjectUpdates,omitempty"`
}
```

**Comment by dougm** [comment]
@thib-ack haven't tried myself, but yes most likely is caused by the omitempty tag.  If removing the tag works for you, we can commit that change.  The tag is there since it is marked optional in the wsdl.  We've had to do similar for a handful of other fields, the wsdl generator has a list of fields to drop the omitempty tag.

**Comment by thib-ack** [comment]
Fixed with #741 
I close the issue
